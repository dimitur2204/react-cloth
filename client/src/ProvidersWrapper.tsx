import React, { PropsWithChildren } from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'

export interface ProvidersWrapperProps {
  store: any
}

const ProvidersWrapper = ({ children, store }: PropsWithChildren<ProvidersWrapperProps>) => (
  <React.StrictMode>
    <Provider store={store}>
      <BrowserRouter>{children}</BrowserRouter>
    </Provider>
  </React.StrictMode>
)

export default ProvidersWrapper
