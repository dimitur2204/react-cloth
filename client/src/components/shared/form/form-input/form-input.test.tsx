import { shallow, ShallowWrapper } from 'enzyme';
import configureStore from 'redux-mock-store';
import ProvidersWrapper from '../../../../ProvidersWrapper';

import FormInput from './form-input.component';

describe('FormInput', () => {
	let wrapper: ShallowWrapper;

	beforeEach(() => {
		const mockStore = configureStore();
		const store = mockStore({});
		wrapper = shallow(
			<ProvidersWrapper store={store}>
				<FormInput />
			</ProvidersWrapper>
		);
	});

	it('should render FormInput component', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
