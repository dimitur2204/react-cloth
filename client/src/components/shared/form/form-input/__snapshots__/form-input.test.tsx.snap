// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput should render FormInput component 1`] = `ShallowWrapper {}`;
